catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/ScriptFragment.kt

```diff
@@ -0,0 +1,1009 @@
+
+    @Transient
+    private var savedLocalUserVariables: List<UserVariable>? = null
+
+    @Transient
+    private var savedLocalLists: List<UserList>? = null
+
+    override fun onCreateActionMode(mode: ActionMode, menu: Menu): Boolean {
+        val inflater = mode.menuInflater
+        inflater.inflate(R.menu.context_menu, menu)
+
+        when (actionModeType) {
+            BACKPACK -> {
+                adapter.setCheckBoxMode(BrickAdapter.SCRIPTS_ONLY)
+                mode.title = getString(R.string.am_backpack)
+            }
+            COPY -> {
+                adapter.setCheckBoxMode(BrickAdapter.CONNECTED_ONLY)
+                mode.title = getString(R.string.am_copy)
+            }
+            DELETE -> {
+                adapter.setCheckBoxMode(BrickAdapter.ALL)
+                mode.title = getString(R.string.am_delete)
+            }
+            COMMENT -> {
+                adapter.selectAllCommentedOutBricks()
+                adapter.setCheckBoxMode(BrickAdapter.ALL)
+                mode.title = getString(R.string.comment_in_out)
+            }
+            NONE -> {
+                adapter.setCheckBoxMode(NONE)
+                actionMode?.finish()
+                return false
+            }
+            CATBLOCKS -> {
+            }
+        }
+        return true
+    }
+
+    override fun onPrepareActionMode(mode: ActionMode, menu: Menu): Boolean = false
+
+    override fun onActionItemClicked(mode: ActionMode, item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.confirm -> handleContextualAction()
+            else -> return false
+        }
+        return true
+    }
+
+    override fun onDestroyActionMode(mode: ActionMode) {
+        resetActionModeParameters()
+        adapter.clearSelection()
+        showBottomBar(activity)
+    }
+
+    private fun handleContextualAction() {
+        if (adapter.isEmpty) {
+            actionMode?.finish()
+        }
+        when (actionModeType) {
+            BACKPACK -> showNewScriptGroupAlert(adapter.selectedItems)
+            COPY -> copy(adapter.selectedItems)
+            DELETE -> showDeleteAlert(adapter.selectedItems)
+            COMMENT -> toggleComments(adapter.selectedItems)
+            NONE -> throw IllegalStateException("ActionModeType not set correctly")
+            CATBLOCKS -> {
+            }
+        }
+    }
+
+    private fun resetActionModeParameters() {
+        actionModeType = NONE
+        actionMode = null
+        adapter.setCheckBoxMode(BrickAdapter.NONE)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        val view = View.inflate(activity, R.layout.fragment_script, null)
+        listView = view.findViewById(android.R.id.list)
+        scriptFinder = view.findViewById(R.id.findview)
+
+        scriptFinder?.setOnResultFoundListener(object : OnResultFoundListener {
```

Comment:
please convert to lambda.
```suggestion
        scriptFinder?.setOnResultFoundListener { sceneIndex, spriteIndex, brickIndex, _, textView ->
```

Reply:
Actually It's creating error in scriptFinder, which was changed in another PR. I synced that PR with this PR so that there won't be error after merging.

Reply:
what kind of Error? could you share it?

Reply:
After converting ScriptFinder in https://github.com/Catrobat/Catroid/pull/4243 and syncing with this PR, I got this error:

![image](https://user-images.githubusercontent.com/46495964/125192813-46aa3200-e267-11eb-91f2-936921600cf4.png)

Reply:
as it suggests, just specify the type of each parameter explicitly.
`sceneIndex: Int, spriteIndex: Int, brickIndex: Int, totalResults: Int, textView: TextView?`